fluent/fluent.go

```diff
@@ -148,27 +169,32 @@ func (f *Fluent) PostWithTime(tag string, tm time.Time, message interface{}) err
 }
 
 func (f *Fluent) EncodeAndPostData(tag string, tm time.Time, message interface{}) error {
-	if data, dumperr := f.EncodeData(tag, tm, message); dumperr != nil {
+	data, dumperr := f.EncodeData(tag, tm, message)
+	if dumperr != nil {
 		return fmt.Errorf("fluent#EncodeAndPostData: can't convert '%s' to msgpack:%s", message, dumperr)
 		// fmt.Println("fluent#Post: can't convert to msgpack:", message, dumperr)
-	} else {
-		f.PostRawData(data)
-		return nil
 	}
+	return f.PostRawDataWithResult(data)
 }
 
 func (f *Fluent) PostRawData(data []byte) {
-	f.mu.Lock()
-	f.pending = append(f.pending, data...)
-	f.mu.Unlock()
-	if err := f.send(); err != nil {
+	if err := f.PostRawDataWithResult(data); err != nil {
+		panic(err)
```

Comment:
PostRawDataWithResult is no panic.
PostRawData is panic.

PostRawData() because public method interface changes dare is not carried out, we have inherited the original implementation.